src/api/app/controllers/status_project_controller.rb

```diff
@@ -0,0 +1,42 @@
+class StatusProjectController < ApplicationController
+  def show
+    dbproj = Project.get_by_name(params[:project])
+    @packages = ProjectStatus::Calculator.new(dbproj).calc_status
+    find_relationships_for_packages(@packages)
+  end
+
+  private
+
+  def role_from_cache(role_id)
+    @rolecache = {}
```

Comment:
I think this is wrong. if you do this in every call... then you are clearing the **cache** every single time. 

This initialization must only happen the first time before the loop in line 32, and this method will use the value memomized if it was calculated before in another iteration of the loop.